### ISSUE TITLE
Muttator (maybe a bit off-topic)

### ISSUE THREAD

**Issue by frenchbeard** [tridactyl/tridactyl#597]
Hi guys,

I don't know if anyone is eager to get on this, but I was wondering which part of current tridactyl's codebase (if any) could be adapted in combination with former [muttator](https://github.com/vimperator/vimperator-labs/tree/master/muttator) to the upcoming Thunderbird 60.

If I had any previous experience with XUL / WebExt, I'd be glad to get on it, but none so far, anyone interested ?

**Comment by bovine3dom** [comment]
You're probably better off making small changes to muttator: https://wiki.mozilla.org/Thunderbird/Add-ons_Guide_57

I don't think Thunderbird supports webextensions yet, so it wouldn't be worth trying to port Tridactyl.

**Comment by frenchbeard** [comment]
@bovine3dom  I'm trying to move out of most of google service, so vmail won't do it for me, but thanks anyway, I'll keep you posted if anything comes up

**Comment by x-ji** [comment]
Seems that we should wait for Thunderbird 63 beta for Webextensions support: https://bugzilla.mozilla.org/show_bug.cgi?id=1396172

**Comment by bovine3dom** [comment]
The docs for "MailExtensions" are here: https://thunderbird-webextensions.readthedocs.io/en/68/

At a glance, it looks like a lot of Tridactyl would need to be rewritten as there aren't clear analogies between the APIs. :man_shrugging:

**Comment by frenchbeard** [comment]
Yep, seems like it. Unfortunately, even though I'd (very much) like to use it, but currently:

-  I wouldn't know where to start from;
- Even though I "hacked" around a couple of things back on Thunderbird up to 52.9, I'm not that familiar with the old bindings (without talking about the new Mailextensions);
- I don't have any experience actually developing in JS, only did a couple of things;
- I won't have any time to spend on it at least until next year.

Would there be anyone keen on getting started with it ?

**Comment by x-ji** [comment]
Meanwhile I can recommend the Thunderbird addon Dorando Keyconfig: https://addons.thunderbird.net/en-US/thunderbird/addon/dorando-keyconfig/

It works up to Thunderbird 60.*, and achieves most of the functionalities that I had been using Muttator for. The only problem is that you can't use it to compose emails. But I normally just write them in my editor before copying them to Thunderbird.

**Comment by j-lakeman** [comment]
I'd like to recommend [External Editor Revived](https://addons.thunderbird.net/en-US/thunderbird/addon/external-editor-revived/) where one can edit emails in external editors such as Vim, Neovim, Emacs, etc.

**Comment by bovine3dom** [comment]
i guess in theory someone motivated enough could do this now because Thunderbird supports webextensions https://webextension-api.thunderbird.net/en/ (they seem to encourage MV3 three rather than MV2, though)
